docs/src/modules/java/pages/actions-publishing-subscribing.adoc

```diff
@@ -0,0 +1,127 @@
+include::ROOT:partial$include.adoc[]
+
+= Publishing and Subscribing with Actions
+
+A very common use case when building Microservices is to publish and subscribe to a stream of events. The source of events can be the journal of an event sourced entity, the value entity state changes or a https://cloud.google.com/pubsub/docs/overview[Google Cloud Pub/Sub] topic for asynchronous messaging between services.
+
+With Actions you can: 
+
+- subscribe to events emitted by an event sourced entity within the same service.
+- subscribe to state changes emitted by a value entity within the same service.
+- subscribe to external events via https://cloud.google.com/pubsub/docs/overview[Google Cloud Pub/Sub].
+- publish events to a Google Cloud Pub/Sub topic.
+
+Messages are guaranteed to be delivered at least once. This means that receivers must be able to handle duplicate messages.
+
+== Publishing Entity events to a Topic
+
+To illustrate how to publish entity events, we will assume the existence of an xref:eventsourced.adoc[Event Sourced] Counter entity that emits events of types: `ValueIncreased` and `ValueDecreased`. We will get the events delivered to an Action, apply some transformation and let them be published to a https://cloud.google.com/pubsub/docs/overview[Google Cloud Pub/Sub] topic. 
+
+include::partial$pubsub-note.adoc[]
+
+[source,protobuf,indent=0]
+.src/main/proto/actions/counter_topic.proto
+----
+include::example$java-eventsourced-counter/src/main/proto/actions/counter_topic.proto[tag=counter-topic]
+----
+<1> Import the Counter Domain file containing the definitions of the events. This is typically a proto definition within the same service.
+<2> The protobuf option (akkaserverless.service) is specific to code-generation as provided by the Akka Serverless Maven plugin. This annotation indicates to the code-generation that an Action must be generated.
+<3> The `Increase` method receives the event `ValueIncreased` and returns the transformed type `Increased`. 
+<4> The `(akkaserverless.method).eventing.in` annotation indicates that events from the entity type `counter` should be delivered to this method (when the type is `ValueIncreased`).
+<5> The `(akkaserverless.method).eventing.out` annotation indicates that the returned value from this method should be published to the topic called `counter-events`.
+
+The class `CounterJournalToTopicAction` gets generated for us based on the proto file defined above.
+
+[source,protobuf,indent=0]
+.src/main/java/com/example/actions/CounterJournalToTopicAction.java
+----
+include::example$java-eventsourced-counter/src/main/java/com/example/actions/CounterJournalToTopicAction.java[tag=counter-topic]
+----
+<1> We convert the incoming domain event `CounterDomain.ValueIncreased` to the outgoing topic API `CounterTopicApi.Increased`.
+<2> We use the converted object to reply.
+
```

Comment:
would be good with a sentence that clarifies that it is the message in the `reply`  that will be published to the topic (when using eventing.out)

Reply:
ok, it's mentioned in the callouts above, but that is slightly wrong in saying returned value

> eventing.out annotation indicates that the returned value from this method should be published to the topic

Reply:
could just repeat it here "<2> We use the converted object as the reply, which will be published to the topic"

Reply:
yep, I will adapt

Reply:
Done in 2322358